### PR TITLE
Suppress memory allocations in Placeholder draw code

### PR DESCRIPTION
**Pull request by kaeawc**
Placeholder is never meant to draw anything so it is fine to ignore and suppress lint warnings about memory allocations in its `onDraw`

### CODE REVIEW

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/Placeholder.java [hunk 1]
```diff
@@ -46,6 +46,9 @@
   private View mContent = null;
   private int mEmptyVisibility = View.INVISIBLE;
 
+  Paint textPaint = new Paint();
+  Rect boundsCache = new Rect();
```
Comment (MGaetan89) [inline constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/Placeholder.java:50]: Why not make these `private`?

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/Placeholder.java [hunk 2]
```diff
@@ -120,22 +123,20 @@ public View getContent() {
...
-      paint.setTextAlign(Paint.Align.CENTER);
-      paint.setTypeface(Typeface.create(Typeface.DEFAULT, Typeface.NORMAL));
-
-      Rect r = new Rect();
-      canvas.getClipBounds(r);
-      paint.setTextSize(r.height());
-      int cHeight = r.height();
-      int cWidth = r.width();
-      paint.setTextAlign(Paint.Align.LEFT);
+      textPaint.setARGB(255, 210, 210, 210);
```
Comment (MGaetan89) [inline constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/Placeholder.java:126]: This never seems to change, so I would set it in the constructor.

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/Placeholder.java [hunk 3]
```diff
@@ -120,22 +123,20 @@ public View getContent() {
...
+      textPaint.setTextAlign(Paint.Align.CENTER);
+      textPaint.setTypeface(Typeface.create(Typeface.DEFAULT, Typeface.NORMAL));
```
Comment (MGaetan89) [inline constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/Placeholder.java:128]: Same for this one.

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/Placeholder.java [hunk 4]
```diff
@@ -120,22 +123,20 @@ public View getContent() {
...
-      paint.setTypeface(Typeface.create(Typeface.DEFAULT, Typeface.NORMAL));
-
-      Rect r = new Rect();
-      canvas.getClipBounds(r);
-      paint.setTextSize(r.height());
-      int cHeight = r.height();
-      int cWidth = r.width();
-      paint.setTextAlign(Paint.Align.LEFT);
+      textPaint.setARGB(255, 210, 210, 210);
+      textPaint.setTextAlign(Paint.Align.CENTER);
```
Comment (MGaetan89) [inline constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/Placeholder.java:127]: I'm not sure, but maybe we could set it to `LEFT` directly, and move it to the constructor?

Comment (kaeawc) [reply constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/Placeholder.java]: I'll make that change but also now realize I should have suppressed this check as its conditionally only in layout preview.

**Review by camaelon — APPROVED** [review]

**Pull request merged**